docs/operations/storage/boltdb-shipper.md

```diff
@@ -83,5 +83,22 @@ Frequency for checking updates can be configured with `resync_interval` config.
 To avoid keeping downloaded index files forever there is a ttl for them which defaults to 24 hours, which means if index files for a period are not used for 24 hours they would be removed from cache location.
 ttl can be configured using `cache_ttl` config.
 
+## Horizontal scaling of non-shared filesystem stores
+
+Using the boltdb-shipper also allows running the single binary Loki (or really just the ingesters) with a `filesystem` object store also using a shared hash ring.
+
+If you configure a shared ring via etcd/consul/memberlist you can run multiple instances of Loki on separate machines with separate filesystems,
+this now works because the ingesters are able to query the store directly.
+
+To enable this configuration in the ingester config you must set `query_store_max_look_back_period` according to how far back you want to store data, or use a value of -1 for infinite.
+
+At query time, any Loki instance can field the query, the instance will then use the ring to ask every other Loki instance for relevant data, 
+and because they ingesters can each query their store as far back as `query_store_max_look_back_period` allows, the correct data can be returned.
+
+Scaling up is as easy as adding more loki instances and letting them talk to the same ring.
+
+Scaling down is possible but manual, you would need to shutdown the loki instance and then physically copy the chunks directory and its index files in their entirety to another Loki instance.
```

Comment:
I think the other instance would also have index files with same name so it would not be just copying of files. I guess we would have to provide a tool to merge boltdb files. We will have to make it clear in the docs until we have that tool.
What do you think?

Reply:
Each ingester should be writing files with a unique name such that you should be able to copy ones file to another without conflicts 

```
uploader := fmt.Sprintf("%s-%d", s.cfg.IngesterName, time.Now().Unix())
```

However, I think I may have found a bug here, this is what I see for my current test ingesters, it looks like the lifecycler ID was empty when they created their files

```
ed@ed-VirtualBox:/tmp$ ls loki1/chunks/index/index_2628
-1589422114
ed@ed-VirtualBox:/tmp$ ls loki2/chunks/index/index_2628
-1589422113
```

These still wouldn't collide because the timestamps were different but they should also have the unique lifecycler ID in front

My suspicion here is they created the files before joining the ring?  Can you take a look at this?

Reply:
Looking at how this works, there is a real chicken and egg problem here when initializing the shipper and trying to get the ingester ID to use in the name file.

I'm not sure it's possible to do this, but maybe you have some ideas?

I'm wondering instead if we should try a different strategy and use the hostname or allow for the name to be configured in the yaml directly?